sdk/communication/azure-communication-callautomation/azure/communication/callautomation/_shared/models.py

```diff
@@ -66,13 +83,21 @@ class CommunicationUserIdentifier(object):
     """
     kind = CommunicationIdentifierKind.COMMUNICATION_USER
 
-    def __init__(self, id, **kwargs):
-        # type: (str, Any) -> None
+    def __init__(self, id: str, **kwargs: Any) -> None:
         self.raw_id = kwargs.get('raw_id', id)
         self.properties = CommunicationUserProperties(id=id)
+        if self.raw_id is None:
+            self.raw_id = _communication_user_raw_id(self)
 
     def __eq__(self, other):
-        return self.__dict__ == other.__dict__
+        try:
+            return self.raw_id == other.properties['id']
+        except TypeError or KeyError or AttributeError:
```

Comment:
This is not the correct syntax, it should be `(TypeError, KeyError, AttributeError)`, though maybe it's just easier to assume any error is `False`...
```suggestion
        except Exception:
```

Reply:
Thanks for the comment! Updated!